typescript/sdk/src/ism/metadata/routing.ts

```diff
@@ -56,3 +57,49 @@ export class RoutingMetadataBuilder implements MetadataBuilder {
     };
   }
 }
+
+export class DefaultFallbackRoutingMetadataBuilder extends RoutingMetadataBuilder {
+  public async build(
+    context: MetadataContext<WithAddress<RoutingIsmConfig>>,
+    maxDepth = 10,
+  ): Promise<string> {
+    const originChain = this.baseMetadataBuilder.multiProvider.getChainName(
+      context.message.parsed.origin,
+    );
+
+    const isRouted = !!context.ism.domains[originChain];
+    if (isRouted) {
+      return super.build(context, maxDepth);
+    }
+
+    if (context.ism.type !== IsmType.FALLBACK_ROUTING) {
+      throw new Error(
+        `Origin domain ${originChain} is not enrolled in DomainRoutingIsm`,
+      );
+    }
+
+    const destinationProvider =
+      this.baseMetadataBuilder.multiProvider.getProvider(
+        context.message.parsed.destination,
+      );
+    const fallbackIsm = DefaultFallbackRoutingIsm__factory.connect(
+      context.ism.address,
+      destinationProvider,
+    );
+    const defaultIsmAddress = await fallbackIsm.module(
+      context.message.parsed.origin,
+    );
```

Comment:
this is not necessarily default ISM right?

Reply:
I can see how this is a bit confusing, but in this flow specifically it is, since its not in the domains mapping, thus implied that the fallbackIsm will return to you the default ISM

Reply:
I see, when `isRouted == false`